### PR TITLE
feat: Add flag stichAsyncCode

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add stichAsyncCode to SentryOptions to be able to enable stitching stack traces
of async code together.

## :bulb: Motivation and Context

Fixes GH-1086

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryCrashAdapter.m [hunk 1]
```diff
@@ -30,6 +31,16 @@ - (BOOL)isBeingTraced
     return sentrycrashdebug_isBeingTraced();
 }
 
+- (void)installAsyncHooks
+{
+    sentrycrash_install_async_hooks();
+}
+
+- (void)deactivateAsyncHooks
```
Comment (marandaneto) [inline Sources/Sentry/SentryCrashAdapter.m:39]: l: why dont we call `uninstallAsyncHooks`? so it matches `installAsyncHooks`

File: Tests/SentryTests/SentryOptionsTest.m [hunk 2]
```diff
@@ -401,6 +401,24 @@ - (void)testInvalidAttachStackTrace
...
+{
+    SentryOptions *options = [self getValidOptions:@{ @"stitchAsyncCode" : @YES }];
+    XCTAssertEqual(YES, options.stitchAsyncCode);
+}
+
+- (void)testInvalidStitchAsyncCode
+{
+    SentryOptions *options = [self getValidOptions:@{ @"attachStacktrace" : @"Invalid" }];
+    XCTAssertEqual(NO, options.attachStacktrace);
+}
```
Comment (marandaneto) [inline Tests/SentryTests/SentryOptionsTest.m:420]: H: you're asserting `attachStacktrace` instead of `stitchAsyncCode`

Comment (philipphofmann) [reply Tests/SentryTests/SentryOptionsTest.m:420]: Good catch thanks

File: Sources/Sentry/SentryCrashAdapter.m (hunk 1 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryCrashAdapter.m:39]: I kept the existing naming. I guess the name is intentionally `deactivate`, see
https://github.com/getsentry/sentry-cocoa/blob/65f4ece762dc954569a7fe71e3249bbd5747e5c6/Sources/SentryCrash/Recording/Tools/SentryHook.c#L298-L300

Comment (marandaneto) [reply Sources/Sentry/SentryCrashAdapter.m:39]: makes sense then

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**